### DISCUSSION TITLE
Support/workaround for raw little-endian RGBA5551 images?

### DISCUSSION THREAD

**Discussion by applecuckoo** [python-pillow/Pillow#8547] [Q&A] [answered]
Hi! I'm currently working on a decoder for a niche texture format. I'm currently using `Image.frombytes` and the raw decoder to convert the image, but there's only a big-endian `raw_mode` for RGBA5551 available. Is there a workaround for this? 

For reference, I've attached a raw 256x256 file and what it should look like. Thank you!
![10_fixed](https://github.com/user-attachments/assets/265c61a1-9879-487e-a008-9b835aa34fa2)
[10.zip](https://github.com/user-attachments/files/17693269/10.zip)

**Comment by radarhere** [comment, marked as answer]
> there's only a big-endian `raw_mode` for RGBA5551 available

Do you mean that you are using the "RGBA;15" unpacker, but rather than RRRRRGGGGGBBBBBA, you would like an unpacker for ABBBBBGGGGGRRRRR? If you're after an immediate solution, I would suggest just reversing the information in your own code first, before passing it to Pillow.

As far as your example goes, I tried running
```python
from PIL import Image
with open("10.dat", "rb") as fp:
    Image.frombytes("RGBA", (512, 512), fp.read(), "raw", "RGBA;15")
```
but received "ValueError: not enough image data", so maybe you're doing something different?

Could you provide some code to demonstrate what you're currently doing, and what you'd like to do?

**Reply by applecuckoo** [reply]
@radarhere About the error - I misremembered, it's 256x256, not 512x512. Yeah, I'll probably reverse the information then. Thank you!

**Reply by Yay295** [reply]
By the way, `RGBA;15` isn't actually "RRRRRGGGGGBBBBBA", it's "ARRRRRGGGGGBBBBB". I created #8158 4.5 months ago to correct this, but it hasn't been merged yet.
